### PR TITLE
Added AddItem feature

### PR DESCRIPTION
**Pull request by msaivasanth**
Added a new feature which adds new products into the website. And made some small changes in other files

### CODE REVIEW

File: src/ProductsContext/ProductState.tsx [hunk 1]
```diff
@@ -56,14 +66,41 @@ const ProductState: React.FC<ProductStateProps> = (props: any) => {
...
+        id: data.length + 1,
+        title: title,
+        description: desc,
+        thumbnail: thumbnail,
+        images: images
+      }
 
-  const value: any = { getProducts, products, setName, setPassword, name, password, handleLogin, handleGetDetails };
+      data.push(newItem)
+      localStorage.setItem('products', JSON.stringify(data))
```
Comment (syedzeeshan1) [inline src/ProductsContext/ProductState.tsx:120]: Also add the call to dummy api add product API. do both for every action. once you get a positive response, update local storage and display from there.

**Comment by msaivasanth** [comment]
Made the suggested changes, please give me the final confirmation to merge with main branch. And I will also make the api call for all the CRUD operations.

File: src/ProductsContext/ProductState.tsx (hunk 1 above)
Comment (msaivasanth) [reply src/ProductsContext/ProductState.tsx:120]: made changes, please give me final confirmation to merge

**Review by syedzeeshan1 — APPROVED** [review]
Give me access to vercel app - syedzeeshan1

**Comment by msaivasanth** [comment]
I have given the access please check.

File: src/Components/AddItem.tsx [hunk 2]
```diff
@@ -2,39 +2,62 @@ import React, { useContext, useState } from 'react';
...
-
-        setTitle('');
-        setDesc('');
-        navigate('/')
-        alert('Product Added successfuly, please scroll down to see added item')
+        if(!title || !desc || !image) {
+            alert('Please Enter all the details.')
+        }
+        else {
+            console.log(title);
```
Comment (syedzeeshan1) [inline src/Components/AddItem.tsx]: nit: remove these

File: src/Components/HomePage.tsx [hunk 3]
```diff
@@ -24,17 +26,18 @@ const HomePage = () => {
...
     <>
+      <ScrollToTop />
       <Navbar />
     <div className='container mt-3'>
         <div className='row'>
         {products.map((product: any) => {
             return <>
-            <div key={product.id} className='col-md-3 my-3 text-center'>
-                <div className="card" style={{width: "20rem"}} >
+            <div key={product.id} className='col-md-3 my-3 text-center' style={{width: "20rem", maxHeight: "500px"}}>
```
Comment (syedzeeshan1) [inline src/Components/HomePage.tsx]: is using inline styling good?

File: package.json [hunk 4]
```diff
@@ -14,7 +14,8 @@
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.22.3",
     "react-scripts": "5.0.1",
-    "typescript": "^5.4.3",
+    "react-scroll-to-top": "^3.0.0",
```
Comment (syedzeeshan1) [inline package.json:17]: package-lock.json should not be a part of this change. put it in .gitignore. Only package.json changes should be good

**Review by syedzeeshan1 — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (msaivasanth) [reply package.json:17]: Sir, I added package-lock.json in  .gitignore but it is not getting ignored

File: src/Components/HomePage.tsx (hunk 3 above)
Comment (msaivasanth) [reply src/Components/HomePage.tsx]: No sir, but in some cases I don't find bootstrap classes for styling.

File: src/Components/AddItem.tsx (hunk 2 above)
Comment (msaivasanth) [reply src/Components/AddItem.tsx]: removed sir.

**Review by syedzeeshan1 — APPROVED** [review]

**Pull request merged**